avocado/utils/asset.py

```diff
@@ -123,7 +123,9 @@ def _download(self, url_obj, asset_path):
             with FileLock(asset_path, 1):
                 shutil.copy(temp, asset_path)
                 self._create_hash_file(asset_path)
-                return self._verify_hash(asset_path)
+                if not self._verify_hash(asset_path):
+                    raise OSError("Hash mismatch, ignoring cache asset.")
```

Comment:
A bit more explicit suggestion for the message: `Hash mismatch, ignoring asset from the cache.`